### ISSUE TITLE
Issues with XML test cases

### ISSUE THREAD

**Issue by chrdebru** [kg-construct/rml-core#104]
- [ ] **No test case for testing selecting values from attributes**
- [ ] Should we have a test case for selecting elements based on conditions?
- [ ] More test cases for language maps? E.g., ensuring no terms are generated when the language map does not yield terms..

**Comment by DylanVanAssche** [comment]
-  No test case for testing selecting values from attributes: +1, feel free to add it if you have one.
-  Should we have a test case for selecting elements based on conditions?: Tricky XPath is welcome I suppose? People do tricky stuff with it so we should check it?
-  More test cases for language maps? E.g., ensuring no terms are generated when the language map does not yield terms.: For now we have the basics, but there might be corner cases left open, so all help is welcome. This is mostly for the maintainers to figure out?

**Comment by chrdebru** [comment]
I agree. I can add a test case for the first if @pmaria and @andimou agree that selecting attributes must use valid XPath queries.

**Comment by chrdebru** [comment]
The first should be solved with #107.

**Comment by dachafra** [comment]
I agree with @chrdebru on both pending tasks. 

I also have many use cases with very tricky XPath expressions, that currently are not supported by many engines.

My question is: do we need them now? If yes, who wants to create them? If not, I'll mark it as `working-group` @DylanVanAssche @chrdebru

**Comment by pmaria** [comment]
> * [ ] Should we have a test case for selecting elements based on conditions?

> I also have many use cases with very tricky XPath expressions, that currently are not supported by many engines.

These would fit best in the io-registry IMHO.

**Comment by DylanVanAssche** [comment]
I agree with @pmaria Tricky XPath expressions need to be specified and tested in rml-io-registry.
We need to collect them first to have a good idea of what is called 'tricky' as that's very vague currently.

**Comment by dachafra** [comment]
>  More test cases for language maps? E.g., ensuring no terms are generated when the language map does not yield terms..

and what about the second? Do we need it? @DylanVanAssche

**Comment by DylanVanAssche** [comment]
> and what about the second? Do we need it?

Depends if the spec requires that if no language tag is generated for the triple while there's a language map, that we don't generate a triple --> basic reference algorithm + spec definition is needed.
